src/Propel/Generator/Reverse/MysqlSchemaParser.php

```diff
@@ -118,6 +130,7 @@ public function parse(Database $database, array $additionalTables = [])
             $this->addPrimaryKey($table);
 
             $this->addTableVendorInfo($table);
+            $this->addDescriptions($table);
```

Comment:
Hmm, let me think about that.

As you already said, this breaks the style of how composition is done in the method. Which I don't even find all that confusing. But even if you do, adding inconsistencies definitely makes it worse. If this change is dear to your heart, I think you need to refactor all the `add...` methods in another PR.

Though I think I even like it better the way it is. Returning an object usually means one of two things: either it is a newly constructed object (the way list functions like map and reduce return a new array), or it is `$this`, so you can chain on it. Neither is true in this case, so it just looks like misunderstood pass-by-reference/pointer. Even worse, if I read something where the same variable gets reassigned several times, I would assume the author was just too lazy to introduce new variables. But if you assign to new variables here (what you explicitly encourage with your interface), they all point to the same object, and the whole thing is just a mess.
To me, it just adds noise, both on a structural level, and literally with just more ascii for what basically are noops. It really seems like a somewhat arbitrary change to the interface, that is supposed to mean something else.

I would argue, that whenever you pass an object, you implicitly pass permission to perform any change the object allows you to make through its interface. But if you want to make it clear to the reader that a function will change the passed object, there are several ways to do it that communicate it more cleanly and clearly:
- Simplest way is through appropriate function names, i.e. `addDescriptionToTable()` or `setTableDescription()` or such.
- You can just not pass on the object, and apply all changes locally: 
```php
$tableDescription = $this->loadTableDescription($table->getName());
$table->setDescription($tableDescription);
```
(This however would hurt readability of the composition function here, which is why I am doing this in another method.)
- You can introduce table builder objects, which is probably the most OOP approach. Then you can just tell the builder to load the description. Most clear, most elegant, most versatile - and probably about three days of work, rewriting the whole class. Not my focus.
- If you really want to use the interface, you can pass the object by alias, i.e. doing `addDescriptions: &$table -> void`. That way, it is clear that the function will do whatever it wants to the object, up to completely replacing it. However, we don't want to replace it, we want to update it. And lots of people will be even more confused.

Thinking about it, I am now convinced that the `addDescriptions()` function should not return the table object. I'll remove that again, and instead suffix the method with "ToTable".
But please, let me know your reasoning. This is obviously important to you, so I am curious about your approach to the issue. I am happy to change my mind.

Reply:
It's not a big deal actually. I agree that it's better to keep everything to the same style.

I like this approach, because IMO it makes code more descriptive and easy to read.
I don't need to check method body to understand that method returns modified object.

But I guess this is matter of taste and conventions :)

Reply:
I think I used to do it that way sometimes. Thanks to the discussion, I am now convinced it is actually wrong, and I would request changes in a review. Funny how that goes. Well, guess I learned something today.

Reply:
This `ToTable` affix is inconsistent with all the other reverse classes. I'd recommend to name it as before (and in all other classes) and not introduce a new naming style.

Reply:
Good point, but too late, it was already merged. Probably needs a new PR, if you feel strongly about it.

Reply:
You can make a new PR here.